lib/oli_web/components/delivery/scored_activities/scored_activities.ex

```diff
@@ -605,7 +629,93 @@ defmodule OliWeb.Components.Delivery.ScoredActivities do
         }
       )
 
-    Repo.all(query)
-    |> hd
+    case Map.get(activity_registrations_mapper, selected_activity.activity_type_id).title do
+      "Multiple Choice" ->
+        activity_attempts =
+          Repo.all(query)
+          |> Repo.preload(:part_attempts)
+
+        frequency_per_choice_id =
+          activity_attempts
+          |> Enum.reduce([], fn activity_attempt, acc ->
+            Enum.reduce(activity_attempt.part_attempts, [], fn part_attempt, pa_acc ->
+              responses =
+                part_attempt.response["input"]
+                |> String.split(" ", trim: true)
+
+              responses ++ pa_acc
+            end) ++
+              acc
+          end)
+          |> Enum.frequencies()
+
+        activity_attempt = hd(activity_attempts)
+
+        choices =
+          activity_attempt.transformed_model["choices"]
+          |> Enum.map(fn choice ->
+            Map.merge(choice, %{"frequency" => Map.get(frequency_per_choice_id, choice["id"])})
```

Comment:
This is not robust against the possibility that choices can be removed in later revisions of an activity.